build.gradle.kts

```diff
@@ -6,43 +6,87 @@ import org.gradle.util.GradleVersion
 import java.nio.file.Files
 import java.nio.file.StandardCopyOption
 import java.text.SimpleDateFormat
-import java.util.*
+import java.util.Date
 
 plugins {
     java
     signing
     `maven-publish`
     idea
-    id("com.adarshr.test-logger") version "2.1.1"
     id("com.diffplug.gradle.spotless") version "3.25.0"
     id("com.gradle.plugin-publish") version "0.12.0"
-    id("io.freefair.lombok") version "4.1.6" apply false
     id("net.linguica.maven-settings") version "0.5"
     id("net.researchgate.release") version "2.8.1"
     id("org.ajoberstar.grgit") version "4.1.0"
-    id("org.kordamp.gradle.jacoco") version "0.42.1"
 
-    if ( org.gradle.util.GradleVersion.current().compareTo(org.gradle.util.GradleVersion.version("6.0")) < 0) {
-        id("com.github.johnrengelman.shadow") version "5.2.0"
+    when {
+        org.gradle.util.GradleVersion.current() < org.gradle.util.GradleVersion.version("5.0") -> {
+            id("com.adarshr.test-logger") version "1.7.1"
+            // XXX: Versions 4.x > 4.0.1 suffer from <https://github.com/johnrengelman/shadow/issues/425>
+            id("com.github.johnrengelman.shadow") version "4.0.1"
+        }
+        org.gradle.util.GradleVersion.current() < org.gradle.util.GradleVersion.version("6.0") -> {
+            id("com.adarshr.test-logger") version "2.1.1"
+            id("com.github.johnrengelman.shadow") version "5.2.0"
+        }
+        org.gradle.util.GradleVersion.current() < org.gradle.util.GradleVersion.version("7.0") -> {
+            id("com.adarshr.test-logger") version "2.1.1"
+            id("com.github.johnrengelman.shadow") version "6.1.0"
+        }
+        else -> {
+            id("com.adarshr.test-logger") version "3.0.0"
+            id("com.github.johnrengelman.shadow") version "7.0.0"
+        }
+    }
 
-    } else {
-        id("com.github.johnrengelman.shadow") version "6.1.0"
+    when {
+        org.gradle.util.GradleVersion.current() < org.gradle.util.GradleVersion.version("5.0") -> {
+            id("io.freefair.lombok") version "2.9.5" apply false
+        }
+        org.gradle.util.GradleVersion.current() < org.gradle.util.GradleVersion.version("5.2") -> {
+            // XXX: Versions 3.x < 3.6.1 suffer from <https://github.com/freefair/gradle-plugins/issues/31>
+            id("io.freefair.lombok") version "3.0.0" apply false
```

Comment:
So this is what causes the lombok.config files - for Gradle 5.0 -> 5.1 ? What about if we only set that ` generateLombokConfig.enabled = false` on appropriate versions ? And ignore any generated files?

Reply:
> So this is what causes the lombok.config files - for Gradle 5.0 -> 5.1 ?

It's Gradle < 5.2 apparently, so 4.10 is included as well.

> What about if we only set that ` generateLombokConfig.enabled = false` on appropriate versions ? And ignore any generated files?

We already disable it on all versions so this applies to 4.10, 5.0, and 5.1 as well.

However, that plugin version ignores this setting and there seems to be no way to turn it off. Creating empty files is a way to work around it creating (invalid) files since the setting does not have any effect.

If there's some way to set it to a file in the root project, then at least we could cut it down to a single file instead of one per module.

Reply:
I have tested on both 4.10 and 5.1 adding ` if (org.gradle.util.GradleVersion.current() >= org.gradle.util.GradleVersion.version("5.2")) { ` around the lombok configuration block ; with that, even with no lombok files it seems to work. It does generate them but I think we could ignore that?

Reply:
I have also added `lombok.config` to `.gitignore`. We never generated the file in the past, but as some versions generate the files now, it's safer.

Reply:
Thats perfect, thanks.